server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -553,22 +555,32 @@ private DataSegment mergeAndPush(final SegmentIdentifier identifier, final Sink
         throw new ISE("Merged target[%s] exists after removing?!", mergedTarget);
       }
 
+      final File mergedFile;
```

Comment:
This code is similar to one in AppenderatorImpl.java, probably we could extract a method to underline the difference.

Reply:
This would be nice to do, but to truly reduce the complexity of the codebase (rather than extracting a method with 10 parameters, that is not simpler) some deeper refactoring is needed, that is out of scope of this PR.